erpnext/shopping_cart/doctype/shopping_cart_settings/shopping_cart_settings.py

```diff
@@ -18,8 +18,12 @@ def onload(self):
 
 	def validate(self):
 		if self.enabled:
+			if self.enable_checkout:
+				self.validate_checkout_gateways()
 			self.validate_exchange_rates_exist()
 
+		self.set_payment_gateway()
```

Comment:
Put this inside the `if self.enable_checkout` check, since that's the only time the gateways will show up anyway